### ISSUE TITLE
Changing content type `config.xml` updates require a restart to take effect

### ISSUE THREAD

**Issue by russdanner** [craftercms/craftercms#6578]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] The issue is in the develop branch
- [X] The issue is in the latest released 4.1.x
- [ ] The issue is in the latest released 4.0.x
- [ ] The issue is in the latest released 3.1.x

### Describe the issue

Modifying a content type `config.xml` requires a Studio restart to take effect.

### Steps to reproduce

Steps:

1. Create a site based on editorial BP
2. Modify the rule for the Category landing page to the following:
```
	<paths>
		<includes>
			<pattern>^/site/website/foo/.*</pattern>
		</includes>
	</paths>
```
3. Save the type
4. Try to create new content under home and note that the Category landing page is still available.
5. Restart and try again. After the restart you will not see the Category landing page

**Comment by jmendeza** [comment]
@russdanner @sumerjabri @rart 
This is happening because `config.xml` is updated using write_content configuration.
Please use `write_configuration` API, so the configuration cache is invalidated properly after the update.
When editing the form, the UI already using `write_configuration`. So a workaround is to update config.xml and then save the content type form.

**Comment by rart** [comment]
@russdanner ready for validation on **4.1.x**.

**Comment by yacdaniel** [comment]
`Verified fixed in 4.2.0-SNAPSHOT`
